### PR TITLE
windows builds + pin snappy

### PR DESCRIPTION
**Pull request by djsutherland**
1. Since there's a C extension built against snappy, python-snappy needs to be pinned to compatible versions of snappy. 1.1.6 and 1.1.7 [are compatible](https://abi-laboratory.pro/tracker/timeline/snappy/) with each other, but not with previous versions (and old versions aren't forward-compatible either).

2. Now that conda-forge has Windows packages, attempt to build there too. We'll see if that works. :)

### CODE REVIEW

**Comment by jakirkham** [comment]
Could we please add the license file while we are at it (i.e. `license_file: LICENSE` in the `about` section)?

**Comment by djsutherland** [comment]
Added the license file, did some other cleanup, added a super-basic correctness test, and also put myself on the maintainers list because why not. :)

**Comment by djsutherland** [comment]
python-snappy doesn't build on VC9: https://github.com/andrix/python-snappy/issues/48, and AppVeyor gave [a similar error](https://ci.appveyor.com/project/conda-forge/python-snappy-feedstock/build/1.0.3/job/544ktk0w21qd7c47). Built successfully on 3.x though.

It has a cffi-based fallback intended for pypy; just added a commit forcing windows py2.7 to fall back to that. If that doesn't work, could also just skip it.

**Comment by jakirkham** [comment]
The failure appears to be caused by VS 2008 not including `stdint.h`. The typical workaround for this situation is to add `msinttypes` as a `build`-only `requirement` for Python 2.7 builds.

Edit: `msinttypes` simply provides a copy of `stdint.h` that should work reasonably well on old Windows compilers.

**Comment by djsutherland** [comment]
Ah, didn't know about that. Will try that tonight, thanks.

**Comment by jakirkham** [comment]
No worries. We should probably add this to the docs.

**Comment by djsutherland** [comment]
`msinttypes` isn't enough, apparently. Will revert to trying to make the cffi version work later.

**Comment by jakirkham** [comment]
It's complaining about the use of `inline` with MSVC 2008. The `inline` keyword is not supported in C89, which is what MSVC 2008 uses for C. Can use `__inline` as a workaround. More details in this [SO answer]( https://stackoverflow.com/a/2765211 ).

Here's how [Blosc fixes this]( https://github.com/Blosc/c-blosc/blob/10fb9a98d0de30a6de301c0b97cbb682a2964f3c/blosc/blosclz.c#L84-L86 ). Might be worth just patching `snappy/crc32c.h` with the same strategy. If that works, would submit it upstream.

**Comment by jakirkham** [comment]
Will need to do the same thing in `snappy/snappymodule.cc` too.

Edit: Appears this includes `snappy/crc32c.h` so should not be necessary.

File: recipe/meta.yaml
Comment (jakirkham) [inline recipe/meta.yaml]: Would actually just apply this patch across the board. This would serve as a good test that it doesn't mess up other Python versions on Windows and works ok on other OSes. As it is all preprocessor statements, this should have no negative effects outside of a tiny bit more compilation time. By doing this, it should also make it easier for upstream to accept the change (and us to rest easy) knowing that it works everywhere just fine. Not that I would expect otherwise. 😉

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks @dougalsutherland. Thoughts @mariusvniekerk and @martindurant?

File: recipe/meta.yaml
Comment (jakirkham) [inline recipe/meta.yaml:16]: xref: https://github.com/andrix/python-snappy/pull/55

File: recipe/meta.yaml
Comment (martindurant) [inline recipe/meta.yaml:48]: Happy to have you onboard

**Comment by martindurant** [comment]
Should there not be a win build appearing (appveyor?)?
Does this require https://github.com/andrix/python-snappy/pull/55 to be merged, or is the patch here identical? (in which case, will merging that cause this recipe to need changing?)

**Comment by djsutherland** [comment]
AppVeyor build is there, but you have to scroll down in the CI box.

Patch is the same as that PR; will just remove it for the next release assuming it goes in.

**Comment by martindurant** [comment]
(oops)

**Comment by jakirkham** [comment]
It is there. Might be buried under the CircleCI builds. So may need to scroll.

The patch here is that PR. It would be great to have that merged. There shouldn't be any further changes near term caused by that patch.

Though we will need to remove the patch from here on the next release (assuming that patch is accepted upstream). Can't apply the same patch twice after all. 😉

**Comment by jakirkham** [comment]
Oops. 🙊 Sorry to dup, @dougalsutherland.

**Comment by martindurant** [comment]
+1

**Comment by djsutherland** [comment]
@mariusvniekerk, thoughts?

**Review by jakirkham — APPROVED** [review]

**Comment by martindurant** [comment]
Who has rights to merge this? I don't :|

**Comment by jakirkham** [comment]
The team update script got broken. 😞 Fortunately we have replaced it with a webservice that is very responsive. If you don't get permissions after I merge this, please let me know.

**Pull request merged**
